compiler/rustc_errors/src/lib.rs

```diff
@@ -864,7 +879,11 @@ impl HandlerInner {
             }
         }
         if diagnostic.is_error() {
-            self.bump_err_count();
+            if matches!(diagnostic.level, Level::Error { lint: true }) {
+                self.bump_lint_err_count();
+            } else {
+                self.bump_err_count();
+            }
         } else {
             self.bump_warn_count();
         }
```

Comment:
at this point, this should probably just be an exhaustive match on `diagnostic.level`

Reply:
Hmm, I was hoping to keep using `is_error` to keep it in sync with `deduplicated_err_count` above.